include/rocksdb/env_encrypt2.h

```diff
@@ -0,0 +1,473 @@
+//  copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+//
+// env_encryption.cc copied to this file then modified.
+
+#ifdef ROCKSDB_OPENSSL_AES_CTR
+#ifndef ROCKSDB_LITE
+
+#include <algorithm>
+#include <cctype>
+#include <iostream>
+
+#include "openssl/aes.h"
+#include "openssl/evp.h"
+#include "openssl/rand.h"
+#include "rocksdb/env_encryption.h"
+#include "util/aligned_buffer.h"
+#include "util/coding.h"
+#include "util/library_loader.h"
+#include "util/random.h"
+
+#endif
+
+namespace ROCKSDB_NAMESPACE {
+
+#ifndef ROCKSDB_LITE
+
+struct Sha1Description {
```

Comment:
Extremely unlikely, but do we need to handle if two keys have same Sha1 hash?

Reply:
In my application code, the second key gets rejected and user has to rename.  Yes, there could be additional code to try a known sample to decrypt and decide which key ... but then that gives the bad guys a known sample too.  I recommend we let this slide.

Reply:
agree. seems to me it is the application responsibility to check it when providing the keys. just asking to confirm.